monai/engines/evaluator.py

```diff
@@ -171,19 +171,16 @@ def _iteration(self, engine: Engine, batchdata: Dict[str, torch.Tensor]) -> Dict
         """
         if batchdata is None:
             raise ValueError("Must provide batch data for current iteration.")
-        inputs, targets = self.prepare_batch(batchdata)
-        inputs = inputs.to(engine.state.device)
-        if targets is not None:
-            targets = targets.to(engine.state.device)
+        inputs, targets, args, kwargs = self.prepare_batch(batchdata, engine.state.device)
```

Comment:
I feel this becomes confusing if the user is already familiar with ignite's prepare batch (returns two items).
could we make it flexible with something like:
```suggestion
        batch = self.prepare_batch(batchdata, engine.state.device)
        if len(batch) == 2:
             inputs, targets = batch
        else:
             inputs, targets, args, kwargs = batch
```
or
```suggestion
        inputs, targets, *args = self.prepare_batch(batchdata, engine.state.device)
```

Reply:
Sure, updated to suggestion 1 in the latest commit.
Thanks.